src/java/org/apache/cassandra/db/compaction/ShardManagerNodeAware.java

```diff
@@ -123,21 +127,38 @@ private Token[] findTokenAlignedSplitPoints(Token[] sortedTokens, int shardCount
                 // pos is -(insertion point) - 1, so calculate the insertion point
                 pos = -pos - 1;
 
-                // Check the neighbors
-                Token leftNeighbor = sortedTokens[pos - 1];
-                Token rightNeighbor = sortedTokens[pos];
-
-                // Choose the nearest neighbor. By convention, prefer left if value is midpoint.
-                if (value.size(leftNeighbor) <= value.size(rightNeighbor))
+                if (pos == 0)
```

Comment:
If the left-side value is already used (i.e. `pos` equals its previous value), I believe we should end up here as well.